### ISSUE TITLE
No matrix operations for sub(SparseMatrixCSC, ..., ...)?

### ISSUE THREAD

**Issue by pkofod** [JuliaLang/julia#15355]
Working on #15354 I realized that
```julia
A = rand(10, 10)
sA = sub(A, 1:10, 1:10)
sA'
```
works, but
```julia
As = sprand(10, 10, 0.3)
sAs = sub(As, 1:10, 1:10)
sAs'
```
does not. I would love to fix it, but I am wondering what the preferred way would be. Should I add a method in `linalg/abstractsparse.jl`, or?

Of course `A'` and `As'` works, but `sA'` seems to be redirected to the correct `ctranspose`, while `sAs'` does not get dispatched to the sparse method.

Edit: So I just realized that this goes for all matrix operations on subs of sparse matrices. Is this intentional? @andreasnoack

**Comment by simonster** [comment]
Related to #14398

**Comment by pkofod** [comment]
Seems both related, and a duplicate, thanks for the ref.

**Comment by timholy** [comment]
Coincidentally, I inadvertently fixed this in #15367.
